mimic/rest/identity_api.py

```diff
@@ -250,6 +250,43 @@ def get_users_details(self, request):
             "updated": seconds_to_timestamp(time.time())
         }))
 
+    @app.route('/v2.0/users/<string:user_id>/OS-KSADM/credentials',
+               methods=['GET'])
+    def get_user_credentials_osksadm(self, request, user_id):
+        """
+        Support, such as it is, for the credentials call.
+
+        reference: http://developer.openstack.org/api-ref-identity-v2-ext.html
+            #listCredentials
+        """
+        if user_id in self.core.sessions._userid_to_session:
+            username = self.core.sessions._userid_to_session[user_id].username
+            password = '9d5ed678fe57bcca610140957afab571'  # echo -n B | md5sum
```

Comment:
md5 is not a great hash algorithm.  Sometimes the presence of the string "md5sum" in a codebase is enough to raise security auditors' hackles.  Also, it's just 32 random hex digits, so it doesn't really matter whether they were the output of `md5sum` or not.

On that note, should they really even be the _same_ magic value every time?  As much as possible I'd like to avoid littering our implementation with magic constants.  You go to some lengths in the assertions below to avoid asserting about a specific value, so `binascii.hexlify(os.urandom(16))` ought to serve this purpose somewhat better than a constant...